lib/Dialect/Calyx/Export/CalyxEmitter.cpp

```diff
@@ -1001,7 +1006,7 @@ void Emitter::emitLibraryPrimTypedByFirstOutputPort(
 void Emitter::emitLibraryFloatingPoint(Operation *op) {
   auto cell = cast<CellInterface>(op);
   // magic number for the index of `left/right` input port
-  size_t inputPortIndex = cell.getInputPorts().size() - 3;
+  size_t inputPortIndex = cell.getInputPorts().size() - 2;
```

Comment:
Question: how come this just "works"? This kind of scares me :)

Reply:
Got it. Can we add an assertion at the very least?

```cpp
auto inputPorts = cell.getInputPorts();
assert inputPortIndex < inputPorts.size();
```

Reply:
sure sounds good!

Reply:
wait, if `size_t inputPortIndex = cell.getInputPorts().size() - 2`, then wouldn't `inputPortIndex` guaranteed to be `< inputPorts.size()`. Did you mean to check `inputPortIndex >= 2` or something like that?

Reply:
Yep sorry typing from a phone, mostly just wanting a bounds assertion so we don't get out of bounds UB case

Reply:
no worries, sure!